### PR TITLE
Fix MyPy Errors for HTTP provider.

### PR DESCRIPTION
**Pull request by kazanzhy**
Fix MyPy Errors for HTTP provider

### CODE REVIEW

File: airflow/providers/http/hooks/http.py [hunk 1]
```diff
@@ -186,8 +186,8 @@ def run_and_check(
 
         # Send the request.
         send_kwargs = {
-            "timeout": extra_options.get("timeout"),
             "allow_redirects": extra_options.get("allow_redirects", True),
+            "timeout": extra_options.get("timeout"),
```
Comment (potiuk) [inline airflow/providers/http/hooks/http.py:189]: Hmm. What was the problem here :) ?

Comment (potiuk) [reply airflow/providers/http/hooks/http.py:189]: Does not seem it fixes anything

Comment (kazanzhy) [reply airflow/providers/http/hooks/http.py:189]: I get the error
```
Argument 2 to "send" of "Session" has incompatible type "**Dict[str, Optional[Any]]"; expected "bool"
```
And these parameters switch helped avoid the error. Very strange

Comment (uranusjr) [reply airflow/providers/http/hooks/http.py:189]: Annotate `send_kwargs` as `Dict[str, Any]` instead.

Comment (kazanzhy) [reply airflow/providers/http/hooks/http.py:189]: Thanks, @uranusjr. It also works. Made changes

**Review by uranusjr — APPROVED** [review]

**Pull request merged**